Startup.cs failiukas iš ExpensesWebApp

ljagintavicius opened the pull request:
Defaultinis startupas su ConnectAndCreate() metodu gale, kuris nuskaito scriptą ir jį paleidžia.

ljagintavicius committed:
Startup.cs failiukas iš ExpensesWebApp

Defaultinis startupas su ConnectAndCreate() metodu gale, kuris nuskaito scriptą ir jį paleidžia.

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,72 @@
+                endpoints.MapControllers();
+            });
+        }
+        public void ConnectAndCreate()
```
Niekur nekvieti šitos funkcijos. Reikia kad ją iškviestų pasileidžiant webappsui.

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,72 @@
+        { 
+        string connetionString;
+        SqlConnection cnn;
+        connetionString = @"Data Source=(LocalDb)\Database;User ID=Laimis;Password=12345";
```
Connection string negalima dėt į source kodą. Configą reikia dėt į appsettings.json  failą, o konkrečių reikšmių išvis commitint negalima, turi tiesiog localiai ir viskas. Nes dabar visi gali pamatyt tavo username/password.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,72 @@
+        string connetionString;
+        SqlConnection cnn;
+        connetionString = @"Data Source=(LocalDb)\Database;User ID=Laimis;Password=12345";
+        var script = File.ReadAllText(@"C:\Users\Ieva\Documents\db.sql");
```
Dabar skaitai iš konkretaus failo. Negerai. Reiktų kad tavo db.sql failas būtų embedded resource.
https://stackoverflow.com/questions/39367349/code-or-command-to-use-embedded-resource-in-visual-studio

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,72 @@
+        connetionString = @"Data Source=(LocalDb)\Database;User ID=Laimis;Password=12345";
+        var script = File.ReadAllText(@"C:\Users\Ieva\Documents\db.sql");
+        var sqlqueries = script.Split(new[] { "GO" }, StringSplitOptions.RemoveEmptyEntries);
+        cnn = new SqlConnection(connetionString);
```
Paskaityk apie IDisposable ir kaip juos reikia naudot kode:
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/using-objects

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,72 @@
+using System;
```
Pridėk į repozitoriją ne tik šitą vieną failiuką, bet visus failus, kurie nėra bin foldery.
Iš esmės kad visi kas pasiimtų tavo sucommitintą kodą, galėtų atsidaryt .sln failiuką su visual studio ir pasileist tavo webappsą.

ljagintavicius committed:
Visi failiukai iš pakoreguoto ExpenseWebApp

Neįsikėlė failiukai iš obj folderio, nežinau ar reikia

ljagintavicius committed:
Pakeitimai pagal komentarus

Pagaliau kažkiek supratau kaip veikia GitHub desktop appsas, nes prieš tai per browserį viską kėliau ^^

amikoliunas commented on .gitignore:
```diff
@@ -0,0 +1,42 @@
+
+bin/Debug/netcoreapp3.1/appsettings.Development.json
```
Nereikia visko vardint. Turėtų užtekt pridėt:
bin/
obj/
.vs
Ir tas turėtų išmest tuos folderius visus

amikoliunas commented on Controllers/ExpenseController.cs:
```diff
@@ -0,0 +1,33 @@
+﻿using System;
```
Per VisualStudio man rodo kad yra nenaudojamų šitų using, pilkai pažymėti. Tau nerodo? Juos galima išmest, nes jų niekam nereikia.

amikoliunas commented on Controllers/ExpenseController.cs:
```diff
@@ -0,0 +1,33 @@
+
+        public ExpenseController (ILogger<ExpenseController> logger)
+        {
+            logger = _logger;
```
Čia turėtų būt _logger = logger;
Dabar kažkoks wtf.

amikoliunas commented on Controllers/ExpenseController.cs:
```diff
@@ -0,0 +1,33 @@
+        [HttpGet]
+        public IEnumerable<expense> Get()
+        {
+            return Enumerable.Range(1, 1).Select(index => new expense
```
Objektus visada vadinam iš didžiosios raidės, tai vadink Expense

amikoliunas commented on ExpensesWebApp.csproj.user:
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Šito failo neišmetei iš brancho

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,80 @@
+            using var conn = new SqlConnection(DatabaseConnectionString);
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            using StreamReader reader = new StreamReader(stream);
+                conn.Open();
```
Čia nereikia indentation papildomo. Beje čia žiauriai kietai, visiškai naujas C# feature, kad nereikia rašyt using(var someting = ...  { something }

amikoliunas commented on Startup.cs:
```diff
@@ -0,0 +1,80 @@
+            var assembly = Assembly.GetExecutingAssembly();
+            var allResourceNames = assembly.GetManifestResourceNames();
+            var resourceName = allResourceNames[0];
+            string DatabaseConnectionString = Configuration.GetConnectionString("MyConnStr");
```
Pakeisk į normalesnį vardą - DatabaseConnectionString

amikoliunas replied:
Turiu omeny tą MyConnStr

amikoliunas commented on obj/Debug/netcoreapp3.1/ExpensesWebApp.AssemblyInfoInputs.cache:
```diff
@@ -0,0 +1 @@
+0a4714d1f887f80169e2e4b35850babb25ab4b33
```
Šitą ir tą dar vieną failą apačioj tai išmesk, jie iš obj folderio.

ljagintavicius committed:
Dar pakeitimų pagal komentarus
